Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift

```diff
@@ -33,4 +58,94 @@ final class SitterProfileViewModelSaveTests: XCTestCase {
         XCTAssertEqual(currentSitter.bio, newBio)
         XCTAssertEqual(currentSitter.pricePerHour, String(newPricePerHour))
     }
+
+    func testSaveMethodCallsUploadMethodWithoutTriggeringError() async {
+        // Given
+        MockURLProtocol.requestHandler = { request in
+            let response = try XCTUnwrap(
+                HTTPURLResponse(
+                    url: XCTUnwrap(request.url),
+                    statusCode: 200,
+                    httpVersion: nil,
+                    headerFields: nil
+                )
+            )
+            return (response, nil)
+        }
+
+        // When
+        await viewModel.save()
+
+        // Then
+        XCTAssertEqual(viewModel.isErrorOccurred, false)
+    }
+
+    func testSaveMethodHandlesErrorStatusCode() async {
+        // Given
+        MockURLProtocol.requestHandler = { request in
+            let response = try XCTUnwrap(
+                HTTPURLResponse(
+                    url: XCTUnwrap(request.url),
+                    statusCode: 400,
+                    httpVersion: nil,
+                    headerFields: nil
+                )
+            )
+            return (response, nil)
+        }
+
+        // When
+        await viewModel.save()
+
+        // Then
+        XCTAssertEqual(viewModel.isErrorOccurred, true)
+    }
+
+    func testAllErrorStatusCodesInResponseTriggersError() async {
+        // Given
+        for statusCode in 400...499 {
+            MockURLProtocol.requestHandler = { request in
+                let response = try XCTUnwrap(
+                    HTTPURLResponse(
+                        url: XCTUnwrap(request.url),
+                        statusCode: statusCode,
+                        httpVersion: nil,
+                        headerFields: nil
+                    )
+                )
+                return (response, nil)
+            }
+
+            // When
+            await viewModel.save()
+
+            // Then
+            XCTAssertEqual(viewModel.isErrorOccurred, true, "Error not triggered for status code \(statusCode)")
+        }
+    }
+
+    func testSaveSendsExpectedNumberOfRequests() async {
+        // Given
+        var requestCount = 0
+
+        MockURLProtocol.requestHandler = { request in
+            requestCount += 1
+
+            let response = try XCTUnwrap(
+                HTTPURLResponse(
+                    url: XCTUnwrap(request.url),
+                    statusCode: 200,
+                    httpVersion: nil,
+                    headerFields: nil
+                )
+            )
+            return (response, nil)
+        }
+
+        // When
+        await viewModel.save()
+
+        // Then
+        XCTAssertEqual(requestCount, 1, "Expected 1 request, but got \(requestCount) requests.")
```

Comment:
Do we really need this text?
```suggestion
        XCTAssertEqual(requestCount, 1)
```

Reply:
Fixed with e9d20b2